### ISSUE TITLE
Portable version

### ISSUE THREAD

**Issue by smaragdus** [ruslanfedoseenko/CuteTorrent#13]
I would like to request a portable version of CuteTorrent which saves all its settings and data inside the program folder.

Someone has already created a portable version:

http://www.thumbapps.org/2016/06/CuteTorrent-libtorrent-BitTorrent-client-portable.html
https://sourceforge.net/projects/thumbapps/files/Internet/CuteTorrent/

but I doubt the  quality of the portabilization. I also don't like PortableApps format. Anyway, I think that a natively portable version of CuteTorrent created by the original developer would be better.

This can be implemented in many different ways, for example:
 - Dedicated portable archive
 - Portable installation mode
 - Start-up parameters
 - Certain configuration files inside CuteTorrent root folder which trigger portable mode (use of root folder for storing of data and settings)
And any of these methods would do and would be satisfactory to me.

Several torrent clients (BitComet, Deluge, Halite, PicoTorrent, Tixati, uTorrent) either have a portable version or can easily be made portable. I will be glad if CuteTorrent joins this group.

Regards

**Comment by ruslanfedoseenko** [comment]
I think that changing settings location is not the only thing to be done.
Could you describe the way you use a portable verison of a torrent?

**Comment by smaragdus** [comment]
Hello, my use of portable torrent clients:

 - A portable torrent client is easier to back-up (all data and settings in one folder)
 - Running a portable torrent client from an external hard drive used for torrents downloads (all the torrents and the torrent client in one place)

**Comment by ruslanfedoseenko** [comment]
Do you ever observe issue i described(with different letters of a hard drive)?

**Comment by smaragdus** [comment]
Sorry, it seems I didn't read your full comment above.
I have an external drive I use for torrents- I connect it to my machine only (its drive letter- "F:\"). Windows remembers the driver letter assigned to different external drives but even if it doesn't one can use the Disk Management to re-assign drive letters.

**Comment by smaragdus** [comment]
**Restrict downloads to removable drive only.**
For me this does not sound very good,

**Automatically change letter in save path to current drive on launch.**
You mean relative paths? This seems to be a better solution.

**Comment by vatterspun** [comment]
> I think that changing settings location is not the only thing to be done.

That's true.  Ultimately the ideal thing is for nothing to happen outside the program folder.  Settings, libraries, dependencies, torrents, downloads, etc. should by default all come from the local folder, where ever it's stored.  

> Could you describe the way you use a portable version of a torrent client?
> What I am thinking about is:
> you put client on a removable device.
> You download torrents to removable device.
> On first machine removable device has path H:\
> On second F:\

It's not just a USB drive situation (and the changing letters).  Many portable users want Cloud drive capability (Dropbox, Google Drive, etc.) as well.  

I personally run all my torrents from inside an encrypted volume (Veracrypt) so anytime settings or other details happen outside of the application folder, it's not private or secure.

**Comment by ruslanfedoseenko** [comment]
My current plans according to information you provided:
Create a launch options (-portable probably) which will make client to use application folder for settings. 
All save paths with this option will be relative to current folder of client. And during adding torrent save path will be restricted to current(volume where client is extracted) volume on a HDD.

**Comment by vatterspun** [comment]
> Create a launch option (-portable probably) which will make client to use application folder for settings.  All save paths with this option will be relative to current folder of client.

That sounds great.  Thanks for that.

> And during adding torrent save path will be restricted to current volume(volume where client is extracted) on a removable device.

By "restricted" do you mean that you can't save torrents to non-removable device folders?  Or just where the default folder is?

My suggestion (for the portable mode) is not to focus specifically on the removable drive, but instead to go with "relative" folders, or always depending on where CuteTorrent is located.

For example, if the program is installed to _X:\CuteTorrent\_, it would write MP3 torrents to _X:\CuteTorrent\Downloads\Music_.  Then, if you move the folder over to the Desktop, it would save to the same folder _C:\Users\NAME\Desktop\CuteTorrent\Downloads\Music_.  That way you can always pick up where you left off no matter where the directory is saved.

In the **Settings**, when you modify folder locations, you might want to add a checkbox to make clear download folders are saving to relative pathways:

![2016-08-08 17_43_43-presentation1 - microsoft powerpoint](https://cloud.githubusercontent.com/assets/1289134/17499049/99092ed6-5d90-11e6-824d-6d7d963f2143.png)

An example of this functionality is available in the excellent DocFetcher: http://i.imgur.com/hGeUqmv.png

**Possible problems**

If the user moves the _CuteTorrent_ folder **without** moving the downloaded files as well, it will disrupt the downloads/seeding.  Some programs I've seen resolve this by prompting users to ask "did you move this file?"  Transmission (for Mac) will just add a yellow "no data found" error icon.
